firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,262 @@
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("a non-delimiter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+Scenario: After output is consumed, ChunkSplitter's behavior resets to that of a ChunkSplitter with an empty internal buffer
+
+  GIVEN("A uint8_t ChunkSplitter completely filled with input data, and no delimiters are consumed with an output call")
+    WHEN("10 bytes of non-delimiter bytes followed by a delimiter byte are passed as input, and output is called between each input")
+      THEN("The input method reports ok status")
+      THEN("The output method reports waiting status")
+      THEN("The input method reports output_ready status")
+      THEN("The output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A chunksplitter constructed with a non-zero delimeter parameter and include_delimeter boolean as false correctly handles input chunks
+
+  GIVEN("A uint8_t chunksplitter constructed with value of delimeter parameter as 0x01 is completely filled with non-delimeter input data")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("The output method is called")
+      THEN("The output method returns waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("the input method reports output ready status on the delimeter byte")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A uint8_t chunksplitter constructed with value of delimeter parameter as 0xff is completely filled with non-delimeter input data")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("The output method is called")
+      THEN("The output method returns waiting status")
+      THEN("The output buffer remains unchanged")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("the input method reports output ready status on the delimeter byte")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A chunksplitter constructed with a non-zero delimeter parameter and include_delimeter boolean as true correctly handles input chunks
```

Comment:
You could merge this scenario with the previous one, and specify the value of `include_delimiter` in the GIVEN, since both of these scenarios are there to test the non-zero delimiter functionality.